package.json

```diff
@@ -36,11 +36,18 @@
     "check": "flow check",
     "dev": "babel-node test/server.js",
     "lint": "eslint src",
+    "precommit": "lint-staged",
     "prepublish": ". ./resources/prepublish.sh",
     "preversion": ". ./resources/checkgit.sh && npm test",
     "test": "npm run lint && npm run check && npm run testonly && npm run build",
     "testonly": "babel-node ./node_modules/.bin/_mocha $npm_package_options_mocha"
   },
+  "lint-staged": {
+    "*.js": [
+      "prettier --jsx-bracket-same-line --single-quote --trailing-comma=all --write",
+      "git add"
```

Comment:
let's also run `npm test` to make sure we didn't break anything, although `prettier` shouldn't have in the first place.